### PR TITLE
CI Configuration Expansion

### PR DESCRIPTION
**Pull request by mathersm-dst**
Attempting to expand the CI builds to include: g++-7, clang-3.9 and a MacOS build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +23,36 @@ matrix:
...
+          packages:
+            - g++-7
+      env: COMPILER=g++-7
+
+    - os: linux
+      compiler: clang
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
```
Comment (jwillemsen) [inline .travis.yml]: clang5 is now there, see ACE/TAO travis ci file

File: .travis.yml [hunk 2]
```diff
@@ -58,16 +76,22 @@ before_script:
...
+         cp -f bin/build/ace/platform_macros.GNU $ACE_ROOT/include/makeinclude/platform_macros.GNU;
+      else
+         echo "Linux and clang detected";
+         cp -f bin/build/ace/platform_macros.GNU.clangTarget $ACE_ROOT/include/makeinclude/platform_macros.GNU;
+      fi
+    fi
 
 script:
   - cd $TAF_ROOT
   - perl $ACE_ROOT/bin/mwc.pl -type gnuace -workers 2 TAF_CI.mwc
```
Comment (jwillemsen) [inline .travis.yml]: Can use a -workers 6 to speed this up a little bit

**Comment by jwillemsen** [comment]
For CIAO/DAnCE I had to add inline=0 and optimize=0 to platform_macros.GNU for travis-ci to keep within the build time limit with gcc7, maybe this is also helpful for LASAGNE

**Comment by mathersm** [comment]
I'll have to take a look at trying that. What kind of a speed up on Travis did you see @jwillemsen ?

**Comment by jwillemsen** [comment]
we got within the time limit, no idea how much faster

**Pull request merged**